### PR TITLE
Analyst

### PR DESCRIPTION
**Pull request by yapollo2002**
first merge to master

### CODE REVIEW

File: src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public NamedParameterJdbcTemplate getJdbcTemplate() {
         return new NamedParameterJdbcTemplate(getDataSource());
     }
 
-    @Bean
+    @Bean ("datasource")
     public DataSource getDataSource() {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java]: call this method ```datasource```

File: src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/v1/coursesatisfaction")
+public class CourseSatisfactionController {
+
+    private final CourseSatisfactionService courseSatisfactionService;
+
+    CourseSatisfactionController(CourseSatisfactionService courseSatisfactionService) {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java:20]: use lombok for this constructor

File: src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+       this.courseSatisfactionService = courseSatisfactionService; }
+
+    @GetMapping("")
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java]: remove ```("")```

File: src/main/java/ita/softserve/course_evaluation_analytics/controller/FeedbackSatisfactionController.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/v1/feedbacksatisfaction")
+public class FeedbackSatisfactionController {
+
+    private final FeedbackSatisfactionService feedbackSatisfactionService;
+
+    FeedbackSatisfactionController(FeedbackSatisfactionService feedbackSatisfactionService) {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/controller/FeedbackSatisfactionController.java:20]: lombok

File: src/main/java/ita/softserve/course_evaluation_analytics/entity/Course.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import lombok.NoArgsConstructor;
+
+import java.util.List;
+
+
+   @Data
+   @NoArgsConstructor
+   @AllArgsConstructor
+public class Course {
+
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/entity/Course.java:14]: format this file

File: src/main/java/ita/softserve/course_evaluation_analytics/mapper/CoursesMapper.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import java.sql.SQLException;
+
+
+
+
+    @Component
+    public class CoursesMapper implements RowMapper<Course> {
+
+        @Override
+        public Course mapRow(ResultSet resultSet, int rowNum) throws SQLException {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/mapper/CoursesMapper.java:18]: reformat file (ctrl+alt+l)

File: src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import javax.sql.DataSource;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Repository
+public class AnalystRepositoryImpl implements AnalystRepository {
+
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private JdbcTemplate jdbcTemplate;
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java:23]: use only ```NamedParameterJdbcTemplate```

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.stereotype.Service;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+
+@Service
+
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java]: remove blank line

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+public class CourseSatisfactionService {
+
+    AnalystRepository analystRepository;
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java:18]: make it private

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+
+    CourseSatisfactionService(AnalystRepository analystRepository) {
+        this.analystRepository = analystRepository;
+    }
+
+    List<CourseFeedbackSatisfaction> coursesFeedbackSatisfaction=new ArrayList<CourseFeedbackSatisfaction>();
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java]: what is the reason for creating this variable at class level?

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java]: remove all these blank lines!

File: src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+import ita.softserve.course_evaluation_analytics.repository.AnalystRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FeedbackSatisfactionService {
+
+      AnalystRepository analystRepository;
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java]: private

File: src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+
+         FeedbackSatisfactionService(AnalystRepository analystRepository) {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java:18]: use lombok

File: src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+                this.analystRepository = analystRepository;
+    }
+
+      List<FeedbackSatisfaction> feedbackSatisfaction=new ArrayList<FeedbackSatisfaction>();
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java]: Never create variables like this

**Review by Forfend — CHANGES_REQUESTED** [review]

File: src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java (hunk 7 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java:23]: some methods for one object is not present in NameParameterJdbcTemplate

File: src/main/java/ita/softserve/course_evaluation_analytics/mapper/CoursesMapper.java (hunk 6 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/mapper/CoursesMapper.java:18]: reformated

File: src/main/java/ita/softserve/course_evaluation_analytics/entity/Course.java (hunk 5 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/entity/Course.java:14]: formated

File: src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java (hunk 3 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/controller/CourseSatisfactionController.java]: removed

File: src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java (hunk 1 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/configuration/SpringConfig.java]: renamed

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java (hunk 9 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java:18]: changed

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java (hunk 10 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java]: moved into method body

File: src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java (hunk 11 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/service/CourseSatisfactionService.java]: removed

File: src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java (hunk 12 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java]: changed

File: src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java (hunk 14 above)
Comment (yapollo2002) [reply src/main/java/ita/softserve/course_evaluation_analytics/service/FeedbackSatisfactionService.java]: moved into method body

File: src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java (hunk 7 above)
Comment (Forfend) [reply src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/AnalystRepositoryImpl.java:23]: ```namedParameterJdbcTemplate.query()``` is what you are looking for

**Review by Forfend — CHANGES_REQUESTED** [review]

**Review by Forfend — APPROVED** [review]

**Pull request merged**